rest/servers/spring/src/main/java/com/connexta/store/rest/spring/StoreApi.java

```diff
@@ -0,0 +1,171 @@
+
+  @ApiOperation(
+      value = "Add metadata to a Dataset.",
+      nickname = "addMetadata",
+      notes = "Clients send information about metadata to add to an existing Dataset.",
+      tags = {
+        "store, promote, addMetadata",
+      })
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "The metadata was added to the Dataset."),
+        @ApiResponse(
+            code = 400,
+            message =
+                "The client message could not be understood by the server due to invalid format or syntax.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 401,
+            message = "The client could not be authenticated.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 403,
+            message = "The client is not authorized.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 404,
+            message = "The requested Dataset could not be found.",
+            response = ErrorMessage.class),
+        @ApiResponse(
+            code = 501,
+            message = "The requested API version is not supported and therefore not implemented.",
+            response = ErrorMessage.class)
+      })
+  @RequestMapping(
+      value = "/dataset/{datasetId}",
+      produces = {"application/json"},
+      consumes = {"application/json"},
+      method = RequestMethod.PUT)
+  @ResponseHeader(
+      name = "Content-Version",
+      description =
+          "The API version used by the server to produce the REST message. The server will accept messages for any minor versions prior to this one.",
+      response = String.class)
+  default ResponseEntity<Void> addMetadata(
+      @ApiParam(
+              value =
+                  "The API version used by the client to produce the REST message. The client must accept responses marked with any minor versions after this one. This implies that all future minor versions of the message are backward compatible with all previous minor versions of the same message.",
+              required = true)
+          @RequestHeader(value = "Accept-Version", required = true)
+          @NotNull
+          @Pattern(regexp = SEMANTIC_VERSION_REGEX)
+          String acceptVersion,
+      @Pattern(regexp = "^[0-9a-zA-Z]+$")
+          @Size(min = 32, max = 32)
+          @ApiParam(value = "The ID of the Dataset.", required = true)
+          @PathVariable("datasetId")
+          String datasetId,
+      @ApiParam(value = "file detail") @Valid @RequestPart("file") MultipartFile file)
+      throws Exception {
+    return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
+  }
+
+  @ApiOperation(
+      value = "Mark a Dataset as quarantined, making the data within the set inaccessible.",
+      nickname = "quarantine",
+      notes =
+          "Clients send metadata information which will be quarantined in addition to the Dataset.",
+      tags = {
```

Comment:
❓ Do these tags serve any purpose?  Are they left over from OpenAPI?  Same question for tags is other files.

Reply:
I think unless we still generate docs from these classes, then there is not much purpose. These classes are largely based on the generated ones by OpenAPI with some modifications.